File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/seek/rebalance/ProductListenerConfiguration.kt
package io.micronaut.kafka.docs.seek.rebalance

import io.micronaut.context.annotation.Requires
import io.micronaut.kafka.docs.Products
import jakarta.inject.Singleton

@Singleton
@Requires(property = "spec.name", value = "ConsumerRebalanceListenerTest")
class ProductListenerConfiguration constructor(producer: ProductClient) {

    init {
        // Records are produced before ProductListener rebalances
        producer.produce(Products.PRODUCT_0)
        producer.produce(Products.PRODUCT_1)
    }
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/seek/rebalance/ProductListenerConfiguration.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.kafka.docs.Products import Products

from .ProductClient import ProductClient


@Singleton
@Requires(property="spec.name", value="ConsumerRebalanceListenerTest")
class ProductListenerConfiguration:

    def __init__(self, producer: ProductClient):
        # Records are produced before ProductListener rebalances
        producer.produce(Products.PRODUCT_0)
        producer.produce(Products.PRODUCT_1)
